src/main/java/de/ait/secondlife/controllers/OfferController.java

```diff
@@ -217,4 +213,118 @@ public ResponseEntity<ResponseMessageDto> recover(
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ResponseMessageDto.class))),
+    })
+    public ResponseEntity<ResponseMessageDto> cancel(@PathVariable Long id) {
+        service.cancelOffer(id);
+        return ResponseEntity.ok(
+                new ResponseMessageDto(String.format("Offer with id <%s> was canceled successfully", id))
+        );
+    }
+
+    @DeleteMapping("/{id}/block-by-admin")
+    @Operation(
+            summary = "Cancel offer by admin",
+            description = "Disabling an Offer. Available only for Admin role and for Offers in statuses: <i>Draft, Verification, Auction started, Auction finished, Qualification</i>."
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ResponseMessageDto.class))),
+            @ApiResponse(responseCode = "403", description = "Forbidden",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ResponseMessageDto.class))),
+            @ApiResponse(responseCode = "404", description = "Offer not found",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ResponseMessageDto.class))),
+    })
+    public ResponseEntity<ResponseMessageDto> block(@PathVariable Long id) {
+        service.blockOfferByAdmin(id);
+        return ResponseEntity.ok(
+                new ResponseMessageDto(String.format("Offer with id <%s> was blocked successfully", id))
+        );
+    }
+
+    @PatchMapping("/{id}/complete")
+    @Operation(
+            summary = "Completing offer",
+            description = "Indication of the winner's bid and transferring an Offer from <i>Qualification</i> status to Completed. Available only for Offer owners and for Offers in <i>Qualification</i> status."
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ResponseMessageDto.class))),
+            @ApiResponse(responseCode = "400", description = "Invalid input",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ValidationErrorsDto.class))),
+            @ApiResponse(responseCode = "403", description = "Forbidden",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ResponseMessageDto.class))),
+            @ApiResponse(responseCode = "404", description = "Offer not found",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ResponseMessageDto.class))),
+    })
+    public ResponseEntity<ResponseMessageDto> complete(
```

Comment:
Maybe we need to return OfferDto to display on UI completed Offer in finished state. It will be convenient option for Frontend to avoid extra request to BD to get entity in finished state